inc/vcf/assembly_report_writer.hpp

```diff
@@ -87,29 +142,33 @@ namespace ebi
 
         ~TextAssemblyReportWriter()
         {
+            file << "Number of matches: " << match_stats.get_num_matches()
+                 << "/" << match_stats.get_num_variants() << std::endl;
+            file << "Percentage of matches: "
+                 << (static_cast<double>(match_stats.get_num_matches()) / match_stats.get_num_variants()) * 100
+                 << "%" << std::endl;
             file.close();
         }
 
-        virtual void write_mismatch(const vcf::RecordCore &record_core) override
+        virtual void match(const vcf::RecordCore &record_core, std::string &match_result) override
+        {
+            match_stats.add_match_result(true);
+        }
+
+        virtual void mismatch(const vcf::RecordCore &record_core, std::string &mismatch_result) override
         {
             match_stats.add_match_result(false);
-            if(report_type == ebi::vcf::INVALID) {
-                file << record_core.line;
-            }
+            file << mismatch_result << std::endl;
```

Comment:
Given that we want errors and warning to look always the same, it would be a better idea to format the message here instead of receiving it as an argument.

Reply:
done, now the mismatch message is formatted within the function. where as I made seperate functionns for generation of warning messages, as there may be different kind of warnings. right now we have two different warnings